lib/mongoid/persistable/maxable.rb

```diff
@@ -0,0 +1,35 @@
+# frozen_string_literal: true
+
+module Mongoid
+  module Persistable
+
+    # Defines behavior for setting a field (or fields) to the larger of
+    # either it's current value, or a given value.
+    module Maxable
+      extend ActiveSupport::Concern
+
+      # Set the given field or fields to the larger of either it's current
+      # value, or a given value.
+      #
+      # @example Set a field to be no less than 100.
+      #   document.max(field: 100)
+      #
+      # @param [ Hash<Symbol | String, Comparable> ] fields The fields to
+      #   set, with corresponding minimum values.
+      #
+      # @return [ Document ] The document.
+      def max(fields)
```

Comment:
The naming here should be consistent with whatever the atomic operator is named. Proposed to be `set_to_ceil/floor`